code/game/machinery/buttons.dm

```diff
@@ -35,122 +35,92 @@
 			pixel_x = 25
 		if(WEST)
 			pixel_x = -25
-	if(SSradio)
-		set_frequency(frequency)
-
-/obj/machinery/driver_button/Initialize(mapload)
-	. = ..()
-	set_frequency(frequency)
 
-/obj/machinery/driver_button/set_frequency(new_frequency)
-	SSradio.remove_object(src, frequency)
-	frequency = new_frequency
-	radio_connection = SSradio.add_object(src, frequency, RADIO_LOGIC)
-	return
-
-/obj/machinery/driver_button/Destroy()
-	if(SSradio)
-		SSradio.remove_object(src, frequency)
-	radio_connection = null
-	return ..()
-
-
-/obj/machinery/driver_button/attack_ai(mob/user as mob)
-	return attack_hand(user)
+/obj/machinery/driver_button/attack_ai(mob/user)
+	attack_hand(user)
 
 /obj/machinery/driver_button/attack_ghost(mob/user)
 	if(user.can_advanced_admin_interact())
-		return attack_hand(user)
+		attack_hand(user)
 
-/obj/machinery/driver_button/attackby(obj/item/W, mob/user as mob, params)
+/obj/machinery/driver_button/wrench_act(mob/user, obj/item/I)
+	. = TRUE
 
-	if(istype(W, /obj/item/detective_scanner))
+	if(!I.tool_use_check(user, 0))
 		return
 
-	if(istype(W, /obj/item/multitool))
-		update_multitool_menu(user)
-		return 1
-
-	if(istype(W, /obj/item/wrench))
-		playsound(get_turf(src), W.usesound, 50, 1)
-		if(do_after(user, 30 * W.toolspeed, target = src))
-			to_chat(user, "<span class='notice'>You detach \the [src] from the wall.</span>")
-			new/obj/item/mounted/frame/driver_button(get_turf(src))
-			qdel(src)
-		return 1
-
-	return ..()
-
-/obj/machinery/driver_button/multitool_menu(mob/user, obj/item/multitool/P)
-	return {"
-	<ul>
-	<li><b>ID Tag:</b> [format_tag("ID Tag","id_tag","set_id")]</li>
-	<li><b>Logic Connection:</b> <a href='?src=[UID()];toggle_logic=1'>[logic_connect ? "On" : "Off"]</a></li>
-	<li><b>Logic ID Tag:</b> [format_tag("Logic ID Tag", "logic_id_tag")]</li>
-	</ul>"}
+	user.visible_message("<span class='notice'>[user] starts unwrenching [src] from the wall...</span>", "<span class='notice'>You are unwrenching [src] from the wall...</span>", "<span class='warning'>You hear ratcheting.</span>")
+	if(!I.use_tool(src, user, 30, volume = I.tool_volume))
+		return
 
-/obj/machinery/driver_button/attack_hand(mob/user as mob)
+	WRENCH_UNANCHOR_WALL_MESSAGE
+	new/obj/item/mounted/frame/driver_button(get_turf(src))
+	qdel(src)
 
+/obj/machinery/driver_button/attack_hand(mob/user)
 	add_fingerprint(usr)
+
 	if(stat & (NOPOWER|BROKEN))
 		return
+
 	if(active)
 		return
+
 	add_fingerprint(user)
```

Comment:
minor nitpick but add_fingerprint is called twice, one here on line 69 and one on line 61. Otherwise, looks pretty good, I'm excited for performance

Reply:
Done